### DISCUSSION TITLE
Dapp project guide

### DISCUSSION THREAD

**Discussion by laradev220** [Cyfrin/foundry-full-course-cu#1072] [Q&A]
hey there! I have chosen a project of sharing system using blockchain. like traditional sharing systems (drive, onedrive), this is going to run on blockchain instead of centralized servers. A university final year project, now i don't know alot of things about web3, dapp etc, i little google search and youtube let me to this awesome guy who is really good at teaching blockchain technology. I completed the first part of this course so far, looking forward to gain more knowledge.
My question is about my project that is it possible to make such a system of course for university assignment, not for productivity? And what technologies should be required to build such an web3 app, mean file sharing using blockchain? Is Foundry framework is enough to build such an app or others thing also be required to learn, like nodejs, vite, javascript frameworks etc?
give your thoughts 
Thanks and kind regards.

**Comment by Niferu** [comment]
Hello @efffii19 

I think you can find most guidence in foundry course for backend, but for fully working Dapp project you will definitely need some front-end skills, so for example JavaScript, React and NextJS for nice front-end. You can also build it using simple JavaScript and HTML, but obviously it will be poor looking. For more details regarding creating whole complete project checkout this: [course](https://github.com/smartcontractkit/full-blockchain-solidity-course-js)
